src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs

```diff
@@ -38764,4 +38732,634 @@
+    {
+        var source = """
+using static AnyClass.AnyEnum.Val;
+
+internal class AnyClass : AnyBaseClass
+{
+    internal interface AnyEnum { }
+}
+
+implicit extension E for AnyClass.AnyEnum
+{
+    public class Val { }
+}
+""";
+        var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+
+        comp.VerifyDiagnostics(
+            // (1,1): hidden CS8019: Unnecessary using directive.
+            // using static AnyClass.AnyEnum.Val;
+            Diagnostic(ErrorCode.HDN_UnusedUsingDirective, "using static AnyClass.AnyEnum.Val;").WithLocation(1, 1),
+            // (1,31): error CS0426: The type name 'Val' does not exist in the type 'AnyClass.AnyEnum'
+            // using static AnyClass.AnyEnum.Val;
+            Diagnostic(ErrorCode.ERR_DottedTypeNameNotFoundInAgg, "Val").WithArguments("Val", "AnyClass.AnyEnum").WithLocation(1, 31),
+            // (3,27): error CS0246: The type or namespace name 'AnyBaseClass' could not be found (are you missing a using directive or an assembly reference?)
+            // internal class AnyClass : AnyBaseClass
+            Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "AnyBaseClass").WithArguments("AnyBaseClass").WithLocation(3, 27));
+
+        var tree = comp.SyntaxTrees.Single();
+        var model = comp.GetSemanticModel(tree);
+        var qualifiedName = GetSyntax<QualifiedNameSyntax>(tree, "AnyClass.AnyEnum.Val");
+        Assert.Equal("E.Val", model.GetSymbolInfo(qualifiedName).Symbol.ToTestDisplayString());
+    }
+
+    [Fact]
+    public void UsingStatic_WithoutErrorBaseType()
+    {
+        var source = """
+using static C.Interface.Nested;
+
+internal class C
+{
+    internal interface Interface { }
+}
+
+implicit extension E for C.Interface
+{
+    public class Nested { }
+}
+""";
+        var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+
+        comp.VerifyDiagnostics(
+            // (1,1): hidden CS8019: Unnecessary using directive.
+            // using static C.Interface.Nested;
+            Diagnostic(ErrorCode.HDN_UnusedUsingDirective, "using static C.Interface.Nested;").WithLocation(1, 1),
+            // (1,26): error CS0426: The type name 'Nested' does not exist in the type 'C.Interface'
+            // using static C.Interface.Nested;
+            Diagnostic(ErrorCode.ERR_DottedTypeNameNotFoundInAgg, "Nested").WithArguments("Nested", "C.Interface").WithLocation(1, 26));
+
+        var tree = comp.SyntaxTrees.Single();
+        var model = comp.GetSemanticModel(tree);
+        var qualifiedName = GetSyntax<QualifiedNameSyntax>(tree, "C.Interface.Nested");
+        Assert.Equal("E.Nested", model.GetSymbolInfo(qualifiedName).Symbol.ToTestDisplayString());
+    }
+
+    [Fact]
+    public void AliasDirective()
+    {
+        var source = """
+using MyNested = C.Interface.Nested;
+
+MyNested.M();
+
+internal class C
+{
+    internal interface Interface { }
+}
+
+implicit extension E for C.Interface
```

Comment:
>C.Interface [](http://example.com/codeflow?start=25&length=11)

It would be good to test a scenario when extension type refers to an alias here in one way or another